src/NServiceBus.Extensions.Hosting/HostBuilderExtensions.cs

```diff
@@ -25,12 +26,27 @@ public static IHostBuilder UseNServiceBus(this IHostBuilder hostBuilder, Func<Ho
                 var endpointConfiguration = endpointConfigurationBuilder(ctx);
                 var startableEndpoint = EndpointWithExternallyManagedContainer.Create(endpointConfiguration, new ServiceCollectionAdapter(serviceCollection));
 
+                if (serviceCollection.Any(x =>
```

Comment:
I tried `Contains` but that didn't really work so I figured LINQ is best.
I've also tried to compare `ImplementationFactory == Factory` but that didn't work either. Given that `NServiceBusHostedService` is an internal type this should be good enough

Reply:
I wonder if a boolean flag in the HostBuilderContext.Properties would be less code and more robust?

Reply:
Implemented your suggestion. Except I just use the presence of the key since that is enough

Reply:
Fair enough, will switch to that.